### DISCUSSION TITLE
How can I assign the apps in the catalog to categories

### DISCUSSION THREAD

**Discussion by junoriosity** [kubeapps/kubeapps#4505] [Q&A] [answered]
When accessing the catalog

<img width="1330" alt="catalog" src="https://user-images.githubusercontent.com/5286536/160151274-9723bd55-5bb7-499f-a6ab-44d6ef2fec97.png">

I see several different categories like "Analytics", "Application server" etc.

I have now three questions:

1. Can I add new categories?
2. Can I somehow control the association of apps to the different categories for existing apps?
3. Is there are way to assign apps after adding a new helm chart repository in the `values.yaml` to the different categories?

**Comment by absoludity** [comment, marked as answer]
Hi @junoriosity .

Sorry to be the bearer of bad news, but I think the answer to all three questions is currently no, since it depends on metadata in the Chart.yaml itself. So yes, you can control it if you are using your own repository and you control the Chart.yaml, but not if you're using a repository published by someone else.

On top of that, there was recent discussions with the Bitnami content team about removing the categories from the Chart.yaml in the Bitnami repository, so the existing categories may disappear at some point (though Carvel packages still support them, irrelevant to your case). Therefore, until we get a chance to re-work the category support to be extra metadata provided in addition to a repo, I don't think what you want will be possible.

**Reply by junoriosity** [reply]
Hi @absoludity ,

many thanks for your reply. Then I would like to ask you two different questions:

- At which part of the code is the panel created?
- What metadata etc. would have to be in the helm chart to be assigned to a category? How does that have to look like?

**Reply by absoludity** [reply]
> Hi @absoludity ,
> 
> many thanks for your reply. Then I would like to ask you two different questions:
> 
>     * At which part of the code is the panel created?

The categories are collected by the plugin and returned with the request to GetAvailablePackageSummaries at https://github.com/kubeapps/kubeapps/blob/v2.4.3/cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go#L242-L247 . You can follow from their to the implementation.

> 
>     * What metadata etc. would have to be in the helm chart to be assigned to a category? How does that have to look like?

You can see examples in the Chart.yaml of the Kubeapps chat itself (or any other chart in the Bitnami catalog, currently) as an annotation in the Chart.yaml: https://github.com/bitnami/charts/blob/master/bitnami/kubeapps/Chart.yaml#L1-L2

**Reply by antgamdia** [reply]
Also, just a side comment (not related to Kubeapps, but the _categories_ issue), in other catalogs, like ArtifactHub, they're using the `keywords` field to add the category(ires): if a keyword matches [a fixed list of categories](https://github.com/artifacthub/hub/blob/master/web/src/utils/data.tsx#L283), then the chart will be assigned to this category.

However, in Kubeapps, we dynamically render the category list based upon this `annotations.category` field as Michael already pointed out.
